src/lib/AutoControlledComponent.js

```diff
@@ -120,10 +131,24 @@ export default class AutoControlledComponent extends Component {
 
   componentWillReceiveProps(nextProps) {
     if (super.componentWillReceiveProps) super.componentWillReceiveProps(nextProps)
+    const { autoControlledProps } = this.constructor
 
-    // props always win, update state with all auto controlled prop
-    const newState = _.pick(nextProps, this.constructor.autoControlledProps)
-    if (!_.isEmpty(newState)) this.setState(newState)
+    // Solve the next state for autoControlledProps
+    const newState = _.transform(autoControlledProps, (acc, prop) => {
+      const isNextUndefined = _.isUndefined(nextProps[prop])
+      const propWasRemoved = _.has(this.props, prop) && isNextUndefined
```

Comment:
I think this needs to be:
```js
const propWasRemoved = _.isUndefined(this.props[prop]) && isNextUndefined
```

If you explicitly pass `undefined` then that `_.has(this.props, prop)` will always return true, which means we'd continually reinitialize to the default value.